### ISSUE TITLE
Rule proposal:  prefer `url.fileURLToPath` over `URL#pathname` for Node

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#2626]
### Description

https://github.com/JounQin/prettier/blob/a15bd00a14ee5f268316976b8a63f9e830a859c4/src/utils/infer-parser.js#L86-L90

### Examples

```js
// ❌
new URL('file:///C:/path/to/file').pathname

// ✅
import { fileURLToPath } from 'node:url'

fileURLToPath(new URL('file:///C:/path/to/file'))
```

### Proposed rule name

prefer-file-url-to-path

**Comment by fisker** [comment]
The idea is good, but it's hard to know that `url` has a `file:` protocol.

**Comment by JounQin** [comment]
Indeed.

**Comment by sindresorhus** [comment]
This should be suggested for https://github.com/eslint-community/eslint-plugin-n instead.
